### PR TITLE
Add go-mod tree-sitter config

### PR DESCRIPTION
**Pull request by camdencheek**
Adds a tree-sitter config for go.mod files as well as highlighting
queries.

A couple of notes:
- This only adds highlighting queries (not indentation, folds, etc.). I couldn't find any documentation on how those are interpreted. Happy to add those as well with some help. 
- This requires a `ftdetect` rule for go.mod files that sets `filetype=gomod`. It seems that graphql also doesn't set its filetype, so I wasn't sure if this is something that should be handled in this plugin. 

Example of it in action:
image

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -104,6 +104,16 @@ list.go = {
   maintainers = {"@theHamsta", "@WinWisely268"},
 }
 
+list.gomod = {
+  install_info = {
+    url = "https://github.com/camdencheek/tree-sitter-go-mod",
+	branch = "main",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:110]: indentation.

**Review by theHamsta — APPROVED** [review]
Very nice work! :tada:

**Review by vigoux — APPROVED** [review]
Bam, simple and efficient, in master it goes, thanks for this contribution!

**Comment by camdencheek** [comment]
Awesome! RE: indents and folds, is there documentation somewhere about how that works? The files look pretty simple, but I'm not confident I fully understand the semantics.

**Comment by theHamsta** [comment]
Folds just enumerates all foldable nodes. I don't fully understand indent. But there should be nodes that cause an indent for their whole length and `@branch` that dedents for this special nodes (e.g. a if/else node would cause an indent but for the line with else you would dedent for this one line.

**Comment by camdencheek** [comment]
Ah, nice. Much simpler than I as expecting. Thanks!

**Pull request merged**